### PR TITLE
status() for SMTPServer

### PR DESCRIPTION
**Pull request by j0code**

### CODE REVIEW

File: src/main.ts [hunk 1]
```diff
@@ -16,10 +16,10 @@ export const sql = new Sequelize({
 // await sql.sync({ alter: true })
 
 // await User.create({
-// 	name: "Cfp",
+// 	name: "Certified Milkshake",
```
Comment (cfpwastaken) [inline src/main.ts:19]:
```suggestion
// 	name: "Cfp",
```

File: src/smtp/status.ts [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+	553: { code: 553, ok: false, message: "Requested action not taken: mailbox name not allowed" },
+	554: { code: 554, ok: false, message: "Transaction has failed" },
+	556: { code: 556, ok: false, message: "Domain does not accept mail" }
+}
+
+const ENHANCED_STATUS_CODES: Record<`${number} ${EnhancedCode}`, EnhancedStatusCode> = {
+	"221 2.0.0":  { code: 221, ok, message: "Goodbye",                                  class: 2, subject: 0, detail: 0 },
+	"235 2.7.0":  { code: 235, ok, message: "Authentication succeeded",                 class: 2, subject: 7, detail: 0 },
+	"432 4.7.12": { code: 432, ok: false, message: "A password transition is needed",          class: 4, subject: 7, detail: 12 },
+	"451 4.4.1":  { code: 451, ok: false, message: "IMAP server unavailable",                  class: 4, subject: 4, detail: 1 },
```
Comment (cfpwastaken) [inline src/smtp/status.ts:73]: We dont have IMAP yet??

**Review by cfpwastaken — CHANGES_REQUESTED** [review]

File: src/main.ts [hunk 3]
```diff
@@ -16,10 +16,10 @@ export const sql = new Sequelize({
...
 // 	username: "cfp",
 // 	password: "1234"
 // })
 
-export const smtpserver = new SMTPServer(25) // Port 25 for regular SMTP, 465 for SMTPS
-const pop3server = new POP3Server(110) // Port 110 for regular POP3, 995 for POP3S
\ No newline at end of file
+export const smtpserver = new SMTPServer(getConfig("smtp_port")) // Port 25 for regular SMTP, 465 for SMTPS
+const pop3server = new POP3Server(getConfig("pop3_port")) // Port 110 for regular POP3, 995 for POP3S
```
[the remark this answers is not in the document]
Comment (j0code) [reply src/main.ts:25]: for testing duh
we can put default port tho
```suggestion
export const smtpserver = new SMTPServer(getConfig("smtp_port"), 25) // Port 25 for regular SMTP, 465 for SMTPS
const pop3server = new POP3Server(getConfig("pop3_port"), 110) // Port 110 for regular POP3, 995 for POP3S
```

File: src/smtp/status.ts (hunk 2 above)
Comment (j0code) [reply src/smtp/status.ts:73]: I added all the common ones.

File: src/main.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (cfpwastaken) [reply src/main.ts:25]: Sure.

**Review by cfpwastaken — APPROVED** [review]
Approved.

**Comment by cfpwastaken** [comment]
Github doesnt seem to like the fact that I made that commit. You gotta make a review too.

**Comment by cfpwastaken** [comment]
I will merge main into this branch now. Do not touch.

File: src/smtp/SMTPServer.ts [hunk 4]
```diff
@@ -40,17 +43,17 @@ export default class SMTPServer {
 					receivingData = false;
 					info.content = info.content.substring(0, info.content.length - 3).replaceAll("\r\n", "\n")
 					await smtpserver.handleNewMail(info)
-					sock.write("250 OK\r\n")
-					logger.log("No longer receiving data -----------------------------------")
+					status(250)
+					console.log("[SMTP] No longer receiving data -----------------------------------")
```
Comment (cfpwastaken) [inline src/smtp/SMTPServer.ts:47]:
```suggestion
					logger.log("[SMTP] No longer receiving data -----------------------------------")
```

**Review by cfpwastaken — CHANGES_REQUESTED** [review]
rest looks good

**Review by cfpwastaken — APPROVED** [review]

**Pull request merged**